src/Components/Card.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace NotificationChannels\Facebook\Components;
+
+use NotificationChannels\Facebook\Exceptions\CouldNotCreateCard;
+use NotificationChannels\Facebook\Exceptions\CouldNotCreateMessage;
+use NotificationChannels\Facebook\Exceptions\CouldNotSendNotification;
+
+
+class Card implements \JsonSerializable
+{
+    /** @var string Card Title */
+    protected $title;
+
+    /** @var string Item Url */
+    protected $image_url;
+
+    /** @var string Image Url */
+    protected $item_url;
+
+    /** @var string Subtitle */
+    protected $subtitle;
+
+    /** @var array Call to Action Buttons */
+    public $buttons = [];
+
+    /**
+     * Create a Card
+     *
+     * @return static
+     */
+    public static function create($title = '')
+    {
+        return new static($title);
+    }
+
+    /**
+     * Create Card constructor
+     */
+    public function __construct($title = '')
+    {
+        $this->title($title);
+    }
+
+    /**
+     * Set Button Title.
+     *
+     * @param $title
+     * @throws CouldNotCreateCard
+     * @return $this
+     */
+    public function title($title)
+    {
+        if (mb_strlen($title) > 80) {
+            throw CouldNotCreateCard::titleLimitExceeded($this->title);
+        }
+        $this->title = $title;
+        return $this;
+    }
+
+    /**
+     * Add up to 3 call to action buttons.
+     *
+     * @param array $buttons
+     *
+     * @return $this
+     * @throws CouldNotSendNotification
+     */
+    public function buttons(array $buttons = [])
+    {
+        if (count($buttons) > 3) {
+            throw CouldNotCreateMessage::messageButtonsLimitExceeded();
+        }
+        $this->buttons = $buttons;
+        return $this;
+    }
+
```

Comment:
If you can make it a trait that would be super helpful.
I don't know of anywhere else off the top of my head we would need it in fb
messaging

On 20 Aug 2016 23:49, "Syed Irfaq R. ⚡️" <notifications@github.com> wrote:

> In src/Components/Card.php
> <https://github.com/laravel-notification-channels/facebook/pull/10#discussion_r75585974>
> :
>
> > +     * Add up to 3 call to action buttons.
> > +     *
> > +     * @param array $buttons
> > +     *
> > +     * @return $this
> > +     * @throws CouldNotSendNotification
> > +     */
> > +    public function buttons(array $buttons = [])
> > +    {
> > +        if (count($buttons) > 3) {
> > +            throw CouldNotCreateMessage::messageButtonsLimitExceeded();
> > +        }
> > +        $this->buttons = $buttons;
> > +        return $this;
> > +    }
> > +
>
> This buttons method seems to be duplicated in another file. Hence, It's
> better we move this to a trait instead. Please let me know if you can do it
> or I'll do it after merging.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/laravel-notification-channels/facebook/pull/10/files/f44a70e62797eeef04e9355223738d90b297d1e9#r75585974>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAw3ikVG5EAf80cewI4nDtWfqkXOBQyHks5qh4STgaJpZM4JpLCo>
> .
>